### ISSUE TITLE
Fix intermittent failure to run

### ISSUE THREAD

**Issue by shrikantgautam-tf** [stylelint/stylelint#6943]
### What steps are needed to reproduce the bug?

Hi,

we have multiple projects and in couple of the projects the check for stylelint is not working and same is happening for our jenkins pipeline.

When we run npm run stylelint
> stylelint
> cross-env-shell stylelint 'ui/**/*.scss' -f verbose --config scripts/stylelint/.stylelintrc.json

It gets stopped immediately after the above line. 

The issue is not consistent across all the projects and local environments it is only happening in the couple of environments and the stylelint versions in those project environments are different.

Is this because of v8-compile-cache issue?

### What Stylelint configuration is needed to reproduce the bug?

```js
{
  "reportNeedlessDisables": true,
  "reportInvalidScopeDisables": true,
  "plugins": ["stylelint-no-unsupported-browser-features", "stylelint-declaration-strict-value"],
  "extends": "stylelint-config-recommended-scss",
  "rules": {
    "color-no-invalid-hex": true,
    "function-calc-no-unspaced-operator": true,
    "function-name-case": "lower",
    "function-url-quotes": "always",
    "length-zero-no-unit": true,
    "string-no-newline": true,
    "unit-no-unknown": true,
    "shorthand-property-no-redundant-values": true,
    "property-no-unknown": true,
    "block-no-empty": true,
    "selector-pseudo-class-no-unknown": true,
    "selector-pseudo-element-colon-notation": "double",
    "selector-type-case": "lower",
    "no-duplicate-selectors": true,
    "no-invalid-double-slash-comments": true,
    "media-feature-name-no-unknown": true,
    "font-family-no-missing-generic-family-keyword": true,
    "no-empty-source": true,
    "property-no-vendor-prefix": true,
    "at-rule-no-vendor-prefix": true,
    "selector-no-vendor-prefix": true,
    "media-feature-name-no-vendor-prefix": true,
    "scss/no-global-function-names": null,
    "no-descending-specificity": null,
    "scss/at-extend-no-missing-placeholder": null,
    "selector-type-no-unknown": null
  }
}
```

### How did you run Stylelint?

In the package.json we have a stylelint which we use.
`"stylelint": "cross-env-shell stylelint 'ui/**/*.scss' -f verbose --config scripts/stylelint/.stylelintrc.json "`

### Which version of Stylelint or dependencies are you using?

As mentioned we have multiple projects and the stylelint version is different in the projects where this issue is happening.
One has stylelint version of 15.4.0 and other has stylelint version 15.3.0.

### What did you expect to happen?

When we run npm run stylelint we expect:

> stylelint
> cross-env-shell stylelint 'ui/**/*.scss' -f verbose --config scripts/stylelint/.stylelintrc.json

4 sources checked
 /ui/sales/src/styles/_variables.scss
 /ui/sales/src/styles/styles.scss
 /ui/sales/src/styles/white/_variables.scss
 /ui/sales/src/styles/white/styles.scss

0 problems found

### What actually happened?

When we run npm run stylelint
> stylelint
> cross-env-shell stylelint 'ui/**/*.scss' -f verbose --config scripts/stylelint/.stylelintrc.json

It gets stopped immediately after the above line.

**Comment by ybiquitous** [comment]
@shrikantgautam-tf Thanks for the report using the template.

> Is this because of v8-compile-cache issue?

Honestly, I'm unsure. Can you reproduce it on your local environment?

**Comment by shrikantgautam-tf** [comment]
Hi @ybiquitous sorry for the delayed response, It happened in the local a couple of days back but the developers made changes to path to test the issue they changed the path as '../**/*.scss' in the below command
> cross-env-shell stylelint '../**/*.scss' -f verbose --config scripts/stylelint/.stylelintrc.json
and we get the error of path being not proper and
then they revert the path back to 'ui/**/*.scss' and it worked and since than in local we are not facing that issue but that can't be done when it comes to pipeline.
I have asked the developers to see if we can reproduce it again and we are trying will send you the image once it reoccurs, but we still facing the issue in the pipeline and in the local we get the same it runs
> stylelint
> cross-env-shell stylelint 'ui/**/*.scss' -f verbose --config scripts/stylelint/.stylelintrc.json
and stops immediately after this without moving further ahead

**Comment by ybiquitous** [comment]
Does the same error happen without `cross-env-shell`?

```shell
stylelint 'ui/**/*.scss' -f verbose --config scripts/stylelint/.stylelintrc.json
```

**Comment by shrikantgautam-tf** [comment]
Yes @ybiquitous even i we remove cross-env-shell it gives the error we tried that and i have attached the image in the comment just now of the issue.

**Comment by ybiquitous** [comment]
Does decreasing target `.scss` files still cause the problem? e.g.

```shell
stylelint 'ui/some/file.scss' -f verbose --config scripts/stylelint/.stylelintrc.json
```

**Comment by shrikantgautam-tf** [comment]
Still the same issue
<img width="1033" alt="MicrosoftTeams-image (29)" src="https://github.com/stylelint/stylelint/assets/95083096/91890b3f-36b7-40b9-a513-8fab7ac0d03e">

**Comment by ybiquitous** [comment]
Thanks. I'll dig into the problem later, but I'd appreciate it if you could provide a minimum reproducible repository or demo (stylelint.io/demo). 🙏🏼

**Comment by jeddy3** [comment]
Closing as inactive. @shrikantgautam-tf Feel free to reopen with a minimum reproducible repo.
